pkg/network/ebpf/c/protocols/http/http.h

```diff
@@ -140,13 +140,14 @@ static __always_inline void http_process(http_transaction_t *http_stack, skb_inf
 
     http->tags |= tags;
 
-    if (http_responding(http)) {
-        http->response_last_seen = bpf_ktime_get_ns();
-    }
-
     if (http_closed(skb_info)) {
         http_batch_enqueue(http);
         bpf_map_delete_elem(&http_in_flight, &http_stack->tup);
+    } else {
+        /* response_last_seen is set by the last payload packet of the response */
+        if (http_responding(http)) {
+            http->response_last_seen = bpf_ktime_get_ns();
+        }
```

Comment:
you need to ensure we don't process empty packets (ack, etc.)

Reply:
`http_allow_packet()` is called in the socket filter earlier and checking `empty_payload`

I added the canary buffer[0] != 0 check but doesn't like much this workaround

Reply:
I move back the `http_responding(http)` to his place before because from the [TCP RFC page 26](https://www.ietf.org/rfc/rfc793.txt) FIN can contain payload, even I don't think linux stack do that


> For sequence number purposes, the SYN is considered to occur before the first actual data octet of the segment in which it occurs, while the FIN is considered to occur after the last actual data octet in a segment in which it occurs.